oat-sa/extension-tao-itemqti
# Fix/TR-3757/Incorrect translations for constraint hints in extended text

Related to: https://oat-sa.atlassian.net/browse/TR-3757

Requires: 
 - [x] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/256
 - [x] once the companion PR has been merged, update the internal dependency (`package*.json` files)

Refactor the constraint hints for the extended text interaction.

Unfortunately, these hints were badly structured, splitting the translation into several chunks by position, each one referring to a single word. Something of the form `{{NUMBER}} __("of") {{NUMBER}} __("characters") __("maximum").`

Because of that, it was impossible to properly translate them as not all languages follow the same words order.

This PR reworks this part, installing the updated client code and providing the translations for several languages with respect to the new constraint hints form: `%s of %s characters maximum.`.

This pull request adds translations for the following languages: Dutch, German, French, Italian, Lithuanian, Portuguese, Spanish. As some of the translations were processed with translation tools, they might need small fixes.

As a lot of files have been changed, you can look at the commits to see atomic changes.

---

**Note:** The list of texts was updated for every language present in the locales. This is needed to synchronise the changes. It should be done each time a text is added, removed, or changed. Unfortunately, it looks like the previous PRs didn't, and several texts have be changed as well.

In the end, not all languages are getting the new translations as per this ticket. The languages which received the translations are only: DE, ES, FR, IT, LT, NL, PT. Please look at the commit [4382beb71df43b2dfc5e8577a998477ff6059f87](https://github.com/oat-sa/extension-tao-itemqti/commit/4382beb71df43b2dfc5e8577a998477ff6059f87) for the accurate diff.

**BREAKING CHANGE:** The translation keys changed from `of`, `characters`, `words`, `maximum`, `recommended` to `%s of %s characters recommended.`, `%s of %s characters maximum.`, `%s of %s words maximum.`

---

How to test:
- checkout the branch
- make sure the frontend code is up to date: `(cd taoQtiItem/views && npm i)`
- update TAO, or at least run the script `tao/scripts/tools/GenerateTranslationBundles.php`
- take a test having items with extended text interactions. The best is to have several of them, each representing a variant of the constraints (a test sample is attached to the ticket)
- check each constraint of the extended text interaction, the text should be correct
- depending on the logged-on user, the text should be translated
